pkg/scaffold/input/input.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Modified from github.com/kubernetes-sigs/controller-tools/pkg/scaffold/input/input.go
+
+package input
+
+// IfExistsAction determines what to do if the scaffold file already exists
+type IfExistsAction int
+
+const (
+	// Overwrite truncates and overwrites the existing file (default)
+	Overwrite IfExistsAction = iota
+
+	// Error returns an error and stops processing
+	Error
+
+	// Skip skips the file and moves to the next one
+	Skip
+)
+
+// Input is the input for scaffoldig a file
+type Input struct {
+	// Path is the file to write
+	Path string
+
+	// IfExistsAction determines what to do if the file exists
+	IfExistsAction IfExistsAction
+	
+	// IsExec indicates whether the file should be written with executable
+	// permissions.
+	// Defaults to false
+	IsExec bool
+
+	// TemplateBody is the template body to execute
+	TemplateBody string
+
+	// Repo is the go project package
+	Repo string
+
+	// ProjectPath is the relative path to the project root
+	ProjectPath string
+
+	// ProjectName is the operator's name, ex. app-operator
+	ProjectName string
```

Comment:
I don't think `Repo`, `ProjectPath` and `ProjectName` are used anywhere.
Remove them if they aren't.

Reply:
I should stop reviewing PRs this late. These are the common fields to be used for all templates. Ignore my comment above.